src/agent.h

```diff
@@ -21,14 +21,16 @@
 #include <memory>
 
 #include "configuration.h"
-#include "reporter.h"
 #include "store.h"
```

Comment:
I believe that this could also be replaced with a forward declaration of MetadataStore here.

Reply:
Not really -- we need the full definition of `MetadataStore` to declare a field of this type, otherwise we'll get a "field has incomplete type" compiler error.
We could, however, forward-declare `MetadataAgentConfiguration`, which I did.